src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import java.time.LocalDate;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+/**
+ * Class used for deadline tasks.
+ */
+class Deadline extends Task {
+
+    /**
+     *  String used to store deadline.
+     */
+    protected LocalDate by;
+
+    /**
+     * Constructor for deadline class.
+     *
+     * @param description Task description.
+     * @param by Deadline.
+     */
+    Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
```

Comment:
Maybe can consider using a more descriptive name that indicates a date/time object?